fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/WorldMixin.java

```diff
@@ -85,7 +85,7 @@ protected boolean onPurgeRemovedBlockEntities(List<BlockEntity> blockEntityList,
 			return blockEntityList.removeAll(removals);
 		}
 
-		// Allocating a set when dealing with huge amounts of loaded block entities and then using removeIf is more performant than List#removeAll
+		// List -> List.removeAll is slower than reallocating the List as a Set and then doing a Set -> List.removalAll when dealing with large amounts of loaded block entities.
```

Comment:
You should really say that it's because `List#contains` is significantly slower than `Set#contains`, which `removeAll` calls.